Kernel/Time/HPETComparator.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <Kernel/Time/HPETComparator.h>
+#include <Kernel/Time/TimeManagement.h>
+
+//#define HPET_COMPARATOR_DEBUG
+
+namespace Kernel {
+
+RefPtr<HPETComparator> HPETComparator::create(u8 number, u8 irq, bool periodic_capable)
+{
+    Function<void(const RegisterState&)> callback = ([&](const RegisterState& regs) { TimeManagement::stale_function(regs); });
+    return adopt(*new HPETComparator(number, irq, periodic_capable, move(callback)));
```

Comment:
This can be written without a temporary:
```cpp
return adopt(*new HPETComparator(number, irq, periodic_capable, [](auto& regs) {
    TimeManagement::stale_function(regs);
}));
```

Also let's not capture `&` unless we need to! The lambda here does not use any captures, so let's not capture anything :)